xlators/mgmt/glusterd/src/glusterd-op-sm.c

```diff
@@ -2821,6 +2823,136 @@ glusterd_set_shared_storage(dict_t *dict, char *key, char *value,
     return ret;
 }
 
+int
+glusterd_compare_bool_value(char *value_1, char *value_2)
+{
+    int ret = -1;
+    gf_boolean_t bool_val_1 = _gf_false, bool_val_2 = _gf_false;
+
+    ret = gf_string2boolean(value_1, &bool_val_1);
+    if (ret == -1) {
+        gf_msg(THIS->name, GF_LOG_ERROR, 0, GD_MSG_INVALID_ENTRY,
+               "Invalid value entry");
+        goto out;
+    }
+
+    ret = gf_string2boolean(value_2, &bool_val_2);
+    if (ret == -1) {
+        gf_msg(THIS->name, GF_LOG_ERROR, 0, GD_MSG_INVALID_ENTRY,
+               "Invalid value entry");
+        goto out;
+    }
+
+    if (bool_val_1 == bool_val_2)
+        ret = 1;
+    else
+        ret = 0;
+
+out:
+    return ret;
+}
+
+int
+glusterd_compare_option_value(glusterd_volinfo_t *volinfo, char *input_key,
+                              char *input_value)
+{
+    struct volopt_map_entry *vme = NULL;
+    int ret = -1;
+    char *def_val = NULL;
+    int keylen;
+    glusterd_conf_t *priv = NULL;
+    xlator_t *this = THIS;
+
+    priv = this->private;
+    GF_VALIDATE_OR_GOTO(this->name, priv, out);
+
+    for (vme = &glusterd_volopt_map[0]; vme->key; vme++) {
+        if (!strcmp(vme->key, input_key)) {
+            /* First look for the key in the priv->opts for global option
+             * and then into vol_dict, if its not present then look for
+             * translator default value */
+            keylen = strlen(vme->key);
+            ret = dict_get_strn(priv->opts, vme->key, keylen, &def_val);
+            if (!def_val) {
+                ret = dict_get_strn(volinfo->dict, vme->key, keylen, &def_val);
+                if (ret == -ENOENT)
+                    def_val = glusterd_get_option_value(volinfo, vme->key);
+                if (!def_val) {
+                    if (vme->value) {
+                        def_val = vme->value;
+                    } else {
+                        ret = glusterd_get_value_for_vme_entry(vme, &def_val);
+                        if (ret)
+                            goto out;
+                        else if (ret == -2)
+                            continue;
+                    }
+                }
+            }
+        }
+    }
```

Comment:
Ack

Reply:
Actually this function should end here. Instead of comparing the _delft_val_ to a given _input_val_, it should forget about _input_val_ and just return or expose _default_val_  (by 'expose' I mean the usual C technique of passing a pointer via a second level pointer and giving a value to this pointer in the function). 

Then of course the function name should be changed, as no comparison is involved. Then this function would be by-and-large ready to be used to implement the above proposal, ie. being dropped into `glusterd_get_default_val_for_volopt`.